tssc/step_implementers/sign_container_image/podman_sign.py

```diff
@@ -0,0 +1,238 @@
+    #   fpr:::::::::DD7208BA0A6359F65B906B29CF4AC14A3D109637:
+    #   grp:::::::::A483EE079EC1D58A954E3AAF3BCC61EDD7596BF0:
+    GPG_IMPORT_FINGER_PRINT_REGEX = re.compile(r"^fpr:+([^:]+):$", re.MULTILINE)
+
+    @staticmethod
+    def step_implementer_config_defaults():
+        """
+        Getter for the StepImplementer's configuration defaults.
+
+        Notes
+        -----
+        These are the lowest precedence configuration values.
+
+        Returns
+        -------
+        dict
+            Default values to use for step configuration values.
+        """
+        return DEFAULT_CONFIG
+
+    @staticmethod
+    def required_runtime_step_config_keys():
+        """
+        Getter for step configuration keys that are required before running the step.
+
+        See Also
+        --------
+        _validate_runtime_step_config
+
+        Returns
+        -------
+        array_list
+            Array of configuration keys that are required before running the step.
+        """
+        return REQUIRED_CONFIG_KEYS
+
+    def _run_step(self):
+        # get the pgp private key to sign the image with
+        image_signer_pgp_private_key = self.get_config_value(
+            'container-image-signer-pgp-private-key'
+        )
+
+        # get the uri to the image to sign
+        push_container_image_step_results = self.get_step_results(DefaultSteps.PUSH_CONTAINER_IMAGE)
+        assert push_container_image_step_results is not None and \
+            'container-image-tag' in push_container_image_step_results, \
+            "Expected key (container-image-tag) to be in step results from step" \
+            f" ({DefaultSteps.PUSH_CONTAINER_IMAGE}): {push_container_image_step_results}"
+        container_image_tag = push_container_image_step_results.get('container-image-tag')
+
+        image_signatures_directory = self.create_working_dir_sub_dir(
+            sub_dir_relative_path='image-signature'
+        )
+
+        # import the PGP key and get the finger print
+        image_signer_pgp_private_key_fingerprint = PodmanSign.__import_pgp_key(
+            pgp_private_key=image_signer_pgp_private_key
+        )
+
+        # sign the image
+        signature_file_path = PodmanSign.__sign_image(
+            pgp_private_key_fingerprint=image_signer_pgp_private_key_fingerprint,
+            image_signatures_directory=image_signatures_directory,
+            container_image_tag=container_image_tag
+        )
+        signature_name = os.path.relpath(signature_file_path, image_signatures_directory)
+
+        return {
+            'container-image-signature-private-key-fingerprint': \
+                image_signer_pgp_private_key_fingerprint,
+            'container-image-signature-file-path': signature_file_path,
+            'container-image-signature-name': signature_name
+        }
+
+    @staticmethod
+    def __import_pgp_key(
+        pgp_private_key
+    ):
+        print("Import pgp private key to sign container image(s) with")
```

Comment:
nit: Consider capitalizing PGP since this is presented to the end-user

Reply:
done